flow_action_components/ExecuteNBAStrategy/force-app/main/default/classes/ExecuteNBAStrategy.cls

```diff
@@ -130,4 +130,30 @@ public with sharing class ExecuteNBAStrategy {
 
     }
 
+    @AuraEnabled
+    public static List<Recommendation> executeNBAFlow(String strategyName, String contextRecordId, Integer maxResults) {
+        Flow.Interview curInterview = Flow.Interview.createInterview(strategyName, new Map<String,Object>{'recordId' =>contextRecordId });
+        curInterview.start();
+        System.debug(curInterview.getVariableValue('outputRecommendations'));
+        return (List<Recommendation>)curInterview.getVariableValue('outputRecommendations');
+    }
+
+
+
+    @AuraEnabled
+    public static string launchFLow(Recommendation rec, String strategyName){
+        FlowDefinitionView flowDef = [SELECT Id, ProcessType, TriggerType, ApiName FROM FlowDefinitionView WHERE apiName = :rec.ActionReference];
+        if(flowDef.ProcessType == 'AutoLaunchedFlow') {
+            Flow.Interview curInterview = Flow.Interview.createInterview(rec.ActionReference, new Map<String, Object>{
+                'recordId' => rec.ExternalId,
+                'strategyName' => strategyName,
```

Comment:
this is only in here because of that recent issue where we need a dummy value, right? let's add a comment to that effect: //a limitation in the recordation interface means that a strategyName is still required to record a reaction even if the recommendation was generated by a flow and not a legacy-style strategy. Use any string that matches an existing strategy (the strategy won't be run and doesn't have to do anything)

Reply:
resolved it